### PR TITLE
Created an add access function with tests and documentation

### PR DESCRIPTION
**Pull request by ErinCain**
I created an add_access function that creates the access element for the EML. This access element has a default public principal and read permission.

### CODE REVIEW

File: R/access-element.R [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#' @title Add Access Element
+#' @description The add access element creates a list that includes all the required
+#' elements for the access section of an EML. Calling add_access() will output the
```
Comment (thesadie) [inline R/access-element.R]: switch "an EML." to "an EML document."

File: R/access-element.R [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#' @title Add Access Element
+#' @description The add access element creates a list that includes all the required
+#' elements for the access section of an EML. Calling add_access() will output the 
+#' default public access permissions. To create a non default access add principal 
+#' and permission values into allow_principal and add_permission.
```
Comment (thesadie) [inline R/access-element.R]: To customize the access level provide values for the `allow_principal` and `add_permission` arguments.

File: R/access-element.R [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#' @title Add Access Element
+#' @description The add access element creates a list that includes all the required
+#' elements for the access section of an EML. Calling add_access() will output the 
+#' default public access permissions. To create a non default access add principal 
+#' and permission values into allow_principal and add_permission. 
+#' @param allow_principal The principal argument refers to who will have access
+#' to this dataset. The allow_principal argument takes in a string. 
+#' With no input the allow_principal will default to "public". To change
```
Comment (thesadie) [inline R/access-element.R:8]: please note the other valid options for principal and permission?

File: R/access-element.R [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    principal = allow_principal
+  } else {
+    principal = "public"
+  }
+  if (!is.null(allow_permission)) {
+    permission = allow_permission
+  } else {
+    permission = "read"
+  }
+  allow = list(principal = principal,
```
Comment (thesadie) [inline R/access-element.R]: i would skip making the allow variable assignment and move the list contents to nested within the access list

**Review by thesadie — CHANGES_REQUESTED** [review]
looks good, some minor changes

File: R/access-element.R (hunk 1 above)
Comment (ErinCain) [reply R/access-element.R]: done.

File: R/access-element.R (hunk 2 above)
Comment (ErinCain) [reply R/access-element.R]: Added.

File: R/access-element.R (hunk 3 above)
Comment (ErinCain) [reply R/access-element.R:8]: Added.

File: R/access-element.R (hunk 4 above)
Comment (ErinCain) [reply R/access-element.R]: Changed.

File: R/access-element.R [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+#' for the allow_principal and add_permission arguments. 
+#' @param allow_principal The principal argument refers to who will have access
+#' to this dataset. The allow_principal argument takes in a string. 
+#' With no input the allow_principal will default to "public". To change
+#' to a different principal please input a new principal. 
+#' 
+#' The valid options for the principal are:
+#'    'public' - special principal indicate that any user or group has a particular access permission 
+#'     and other individual users or groups can be specified if they are defined in the 
+#'     "https://pasta.edirepository.org/authentication" authentication system
```
Comment (thesadie) [inline R/access-element.R]: use the appropriate roxygen syntax to make URL clickable

**Review by thesadie — CHANGES_REQUESTED** [review]

File: R/access-element.R [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+#' elements for the access section of an EML document. Calling add_access() will output the 
+#' default public access permissions. To customize the access level provide values 
+#' for the allow_principal and add_permission arguments. 
+#' @param allow_principal The principal argument refers to who will have access
+#' to this dataset. The allow_principal argument takes in a string. 
+#' With no input the allow_principal will default to "public". To change
+#' to a different principal please input a new principal. 
+#' 
+#' The valid options for the principal are:
+#'    'public' - special principal indicate that any user or group has a particular access permission
```
Comment (thesadie) [inline R/access-element.R]: also, the indentation is not rendered in the docs when viewed within rstudio.

File: R/access-element.R [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+#' 
+#' @param allow_permission The permission argument refers to the type of access 
+#' that is allowed to the principal. The allow_permission argument takes in a string. 
+#' With no input the allow_permission will default to "read". This default permission
+#' allows the principal to read the dataset but not edit. To change to a different 
+#' permission please input a new permission. 
+#' 
+#' The valid options for permission are: 
+#'   'read' - allow viewing of the resource, 
+#'   'write' - allow modification of the resource (except for access rules),
```
Comment (thesadie) [inline R/access-element.R]: same here, if you want to have it indented research how to have that formatting

**Review by thesadie — CHANGES_REQUESTED** [review]

File: R/access-element.R (hunk 7 above)
Comment (ErinCain) [reply R/access-element.R]: Added list formatting.

File: R/access-element.R (hunk 6 above)
Comment (ErinCain) [reply R/access-element.R]: Added list formatting.

File: R/access-element.R (hunk 5 above)
Comment (ErinCain) [reply R/access-element.R]: Fixed URL.

**Pull request merged**